tests/subsys/logging/log_core_additional/log_sync.conf

```diff
@@ -1,5 +1,6 @@
 CONFIG_MAIN_THREAD_PRIORITY=5
 CONFIG_ZTEST=y
+CONFIG_ZTEST_NEW_API=y
```

Comment:
You don't need CONFIG_ZTEST_NEW_API in log_sync.conf and log_user.conf as it is already specified in prj.conf.

Reply:
If There is no CONFIG_ZTEST_NEW_API in log_sync.conf and log_user.conf , build failure will occur on logging.add.log_user and logging.add.sync:
```
zephyrproject/zephyr/tests/subsys/logging/log_core_additional/src/log_test.c:479:1: error: return type defaults to 'int' [-Werror=implicit-int]
  479 | ZTEST_USER(test_log_core_additional, test_log_msg_create_user)
      | ^~~~~~~~~~
zephyrproject/zephyr/tests/subsys/logging/log_core_additional/src/log_test.c: In function 'ZTEST_USER':
zephyrproject/zephyr/tests/subsys/logging/log_core_additional/src/log_test.c:479:1: error: type of 'test_log_core_additional' defaults to 'int' [-Werror=implicit-int]
zephyrproject/zephyr/tests/subsys/logging/log_core_additional/src/log_test.c:479:1: error: type of 'test_log_msg_create_user' defaults to 'int' [-Werror=implicit-int]
zephyrproject/zephyr/tests/subsys/logging/log_core_additional/src/log_test.c: At top level:
zephyrproject/zephyr/tests/subsys/logging/log_core_additional/src/log_test.c:523:38: error: expected ')' before '(' token
  523 | ZTEST_SUITE(test_log_core_additional, NULL, test_log_core_additional_setup, NULL, NULL, NULL);
      |                                      ^
      |                                      )
zephyrproject/zephyr/tests/subsys/logging/log_core_additional/src/log_test.c: In function 'ZTEST_USER':
zephyrproject/zephyr/tests/subsys/logging/log_core_additional/src/log_test.c:500:1: error: control reaches end of non-void function [-Werror=return-type]
  500 | }
      | ^
At top level:
zephyrproject/zephyr/tests/subsys/logging/log_core_additional/src/log_test.c:515:14: error: 'test_log_core_additional_setup' defined but not used [-Werror=unused-function]
  515 | static void *test_log_core_additional_setup(void)
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

```